### PR TITLE
update php-cs-fixer to v3 + tweak the configuration + apply it to the codebase

### PR DESCRIPTION
**Pull request by lyrixx**

### CODE REVIEW

File: tests/Jane/JaneTest.php [hunk 1]
```diff
@@ -40,7 +47,8 @@ public function testCreateIndexAndSearchWithJaneObject()
         ];
         $encoders = [new JsonEncoder(
             new JsonEncode(),
-            new JsonDecode([JsonDecode::ASSOCIATIVE => true])),
+            new JsonDecode([JsonDecode::ASSOCIATIVE => true])
```
Comment (welcoMattic) [inline tests/Jane/JaneTest.php:50]: Bizarre ça, non ?

File: tests/Transport/HttpClientTransportTest.php [hunk 2]
```diff
@@ -58,21 +65,22 @@ public function testCreateIndex(): void
...
+                      "hits" : {
+                        "total" : {
+                          "value" : 0,
+                          "relation" : "eq"
+                        },
+                        "max_score" : null,
+                        "hits" : [ ]
+                      }
+                    }
+                    JSON
```
Comment (welcoMattic) [inline tests/Transport/HttpClientTransportTest.php:82]: IIRC, on Windows it could break the test. Do we need to care about?

Comment (lyrixx) [reply tests/Transport/HttpClientTransportTest.php:82]: I don't understand :/ why would it breaks test?

**Comment by lyrixx** [comment]
Failure are not related. There are all deprecation notice.

**Comment by lyrixx** [comment]
There are too much deprecation + type error. This lib does not seem compatibly with something.

**Pull request merged**